### DISCUSSION TITLE
16bit float?

### DISCUSSION THREAD

**Discussion by vurt72** [neonbjb/tortoise-tts#266] [Q&A] [answered]
How can i save my .wav's as 16bit float? i've never seen this format before, not something that is available in any audio program that i've ever ran into, 64 and 32 are standard.
Anyone found a converter? 

Cheers

**Comment by sbersier** [comment]
The resulting audio is indeed encoded as mono, pcm_f32le at 24000 Hz. Which is not a real problem nowadays since most players (VLC, MPlayer, ...) can play it.
But you can re-encode it with, for example, FFmpeg (freely available for linux, Mac and Windows).
If you want a CD format (16 bits / 44100 Hz):
`ffmpeg -i <the_produced_audio_file.wav> -ar 44100 -q:a 0 <new_filename.wav>`

You can check the result with `ffprobe <audio_file>`  (check for Stream #0:0:)

**Comment by vurt72** [comment]
Ah, resulting, i was referring to "Save the clips as a WAV file with floating point format and a 22,050 sample rate."
I could swear it said "16 bit float", or it says that on another page i was on.. hmm. well anyways, this does not seem to matter from more experiments, i save my voice .wav's as PCM and it seems to handle them fine. But maybe there's an advantage of using float instead, like rendering time, didn't compare.

Really amazing results when it works, been using and having fun with text-to-speech since the C64 days! What at time it has been the last year with all the new AI stuff! :)

**Comment by sbersier** [comment]
Yeah, the results are absolutely great and things are moving really fast!

**Comment by sbersier** [comment, marked as answer]
Ah! Now, I think I understand your question... You were referring to the audio samples needed in order to generate the voice latents, right? (cf. "Adding a new voice" paragraph on the landing page). 
The bit depth and format (s16 ,f32) and the audio rate don't matter as long as it is wav or mp3. The "load_audio" function (in tortoise-tts/tortoise/utils/audio.py between lines 29 and 55) will handle it and resample it at 22050 Hz. You can give 48kHz f32le wav or 96kHz mp3 files and it should still work.

**Reply by vurt72** [reply]
Thanks! Good to know!

**Reply by FurkanGozukara** [reply]
@sbersier i dont want it to convert

so 32 fp 22050 wav wouldnt be converted is that native support?

**Reply by sbersier** [reply]
If you look into tortoise-tts/tortoise/utils/audio.py in lines 16-55:
You can see that tortoise accepts: int16, int32, float16, float32 (lines 16-26) , WAV and mp3 files (lines 30-36) . If it is not 22050, it will resample it at that rate (line 47). So you should be OK, nothing to do.

**Reply by sbersier** [reply]
Note: line26 reads:
 `   return (torch.FloatTensor(data.astype(np.float32)) / norm_fix, sampling_rate) `

So, float32 seems to be what TorToiSe will use ultimately. But I'am not the author of tortoise...
